src/Converter/Extensions/InstagramConverter.php

```diff
@@ -0,0 +1,71 @@
+<?php
+
+namespace Predmond\HtmlToAmp\Converter\Extensions;
+
+use League\Event\EventInterface;
+use Predmond\HtmlToAmp\ElementInterface;
+use Predmond\HtmlToAmp\Converter\ConverterInterface;
+
+class InstagramConverter implements ConverterInterface
+{
+    public function hasInstagram(ElementInterface $element)
+    {
+        $classes = explode(' ', $element->getAttribute('class'));
+        return in_array('instagram-media', $classes);
+    }
+
+    public function findTheCode($element) 
+    {
+        $href = $element->getAttribute('href');
+        if ($href != '') {
+            $parts = explode('/', $href);
```

Comment:
I think a regex would be more consistent here:

```php
if (1 === preg_match('/(?:instagr\.am|instagram\.com)\/p\/([^\/]+)\/?$/i', $href, $matches)) {
    return $matches[1];
}
```

See this demo http://regexr.com/3cov4 and see this https://github.com/xentek/instagram-embed/blob/master/instagram-embed.php#L20